client/state/selectors/is-domain-sidebar-experiment-user.js

```diff
@@ -0,0 +1,14 @@
+import { getCurrentUser } from 'calypso/state/current-user/selectors';
+import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
+
+/**
+ * Whether the user is in the Domain sidebar upsell experiment.
+ *
+ * @returns {boolean} Whether the user is in the Domain sidebar upsell experiment.
+ */
+export const isDomainSidebarExperimentUser = ( state ) => {
+	const currentUser = getCurrentUser( state );
+	const domainAndPlanPackage = getCurrentQueryArguments( state ).domainAndPlanPackage;
```

Comment:
`getCurrentQueryArguments` could be `null`, and trying to access a property on a `null` will trigger a `TypeError`. We could use optional chaining to address that.

Furthermore, we're using this query argument value as a boolean, so let's make sure to cast it. It could theoretically contain anything - including falsy values, so casting here would be required. We can prefix with `!!` to cast it to a boolean.

```suggestion
	const domainAndPlanPackage = !! getCurrentQueryArguments( state )?.domainAndPlanPackage;
```

Reply:
Interesting to know, [fixed here](https://github.com/Automattic/wp-calypso/pull/73017/commits/3aeff691fe16323df3aa05089b1791e5309eff8b).